tests/backends/test_backend.py

```diff
@@ -117,3 +118,97 @@ def test_environment_board_get_backend_unknown() -> None:
     environment = MockEnvironment
     with pytest.raises(NotImplementedError):
         assert environment.get_backend(Mock2Board)
+
+
+def test_backend_check_has_environment() -> None:
+    """Test that an error is thrown if a backend is defined without an environment."""
+    with pytest.raises(ValueError):
+        class NoEnvironmentBackend(Backend):
+            """I have no environment."""
+
+            board = MockBoard
+
+            @classmethod
+            def discover(cls) -> Set['Board']:
+                return set()
+
+            def get_firmware_version(self) -> Optional[str]:
+                return None
+
+
+def test_backend_check_multiple_backends_same_env() -> None:
+    """Test that we can't define two backends for the same board/environment combo."""
+    test_environment = Environment("test_environment")
+
+    with pytest.raises(RuntimeError):
+        class BackendOne(Backend):
+            """I have no environment."""
```

Comment:
This docstring doesn't match the rest of the class (but I don't think it needs a docstring anyway).